web/components/roomConfig/GitlabRepoConfig.tsx

```diff
@@ -0,0 +1,167 @@
+    }, [onPicked, results, filter, setCurrentProjectPath]);
+
+
+
+    return <div>
+        {instances === null && <p> Loading GitLab instances. </p>}
+        {instances?.length === 0 && <p> You are not logged into any GitLab instances. </p>}
+        <InputField visible={instances?.length > 0} label="GitLab Instance" noPadding={true}>
+            {/*TODO: on change required here*/}
+            <select>
+                {instances?.map(i => <option key={i.name}>{i.name}</option>)}
+            </select>
+        </InputField>
+        <InputField visible={instances?.length > 0} label="Project" noPadding={true}>
+            <small>{currentProjectPath ?? ""}</small>
+            <input onChange={updateSearchFn} list="gitlab-projects" type="text" />
+            <datalist id="gitlab-projects">
+                {results?.map(i => <option path={i.state.path} value={i.name} key={i.name} />)}
+            </datalist>
+        </InputField>
+    </div>;
+}
+
+const ConnectionConfiguration: FunctionComponent<ConnectionConfigurationProps<unknown, GitLabRepoResponseItem, GitLabRepoConnectionState>> = ({api, existingConnection, onSave, onRemove }) => {
+    const [ignoredHooks, setIgnoredHooks] = useState<string[]>(existingConnection?.config.ignoreHooks || []);
+    const setIgnoredHook = useCallback(evt => {
+        const key = (evt.target as HTMLElement).getAttribute('x-event-name');
+        if (ignoredHooks.includes(key)) {
+            setIgnoredHooks(ignoredHooks.filter(k => k !== key));
+        } else {
+            setIgnoredHooks([...ignoredHooks, key]);
+        }
+    }, [ignoredHooks, setIgnoredHooks]);
+    const [newInstanceState, setNewInstanceState] = useState<GitLabRepoConnectionState|null>(null);
+
+
+
+    const canEdit = !existingConnection || (existingConnection?.canEdit ?? false);
+    const commandPrefixRef = createRef<HTMLInputElement>();
+
+    const onSaveCb = useCallback(() => {
+        onSave({
+            ...(existingConnection?.config || newInstanceState),
+            ignoreHooks: ignoredHooks as any[],
+            commandPrefix: commandPrefixRef.current.value,
+        });
+    }, [existingConnection, newInstanceState, ignoredHooks, commandPrefixRef, onSave]);
+    
+    return <div>
+        {!existingConnection && <ConnectionSearch api={api} onPicked={setNewInstanceState} />}
+        <InputField visible={!!existingConnection} label="GitLab Instance" noPadding={true}>
+            <input disabled={true} type="text" value={existingConnection?.config.instance} />
+        </InputField>
+        <InputField visible={!!existingConnection} label="Project" noPadding={true}>
+            <input disabled={true} type="text" value={existingConnection?.config.path} />
+        </InputField>
+        <InputField visible={!!existingConnection || !!newInstanceState} ref={commandPrefixRef} label="Command Prefix" noPadding={true}>
+            <input type="text" value={existingConnection?.config.commandPrefix || "!gl"} />
```

Comment:
Could empty string be a valid existing configuration for the Command Prefix?

Reply:
Empty string is indeed valid, I'll relegate "!gl" as a placeholder.